### PR TITLE
vim: Add option to build vim with gtk2 gui

### PR DESCRIPTION
**Pull request by paulhybryant**
Add option to build vim with gtk2 gui.

### CODE REVIEW

File: Library/Formula/vim.rb [hunk 1]
```diff
@@ -33,6 +34,7 @@ class Vim < Formula
   depends_on "luajit" => :optional
   depends_on :x11 if build.with? "client-server"
   depends_on "homebrew/dupes/ncurses" unless OS.mac?
+  depends_on "gtk+" if build.with? "gui-gtk2"
```
Comment (sjackman) [inline Library/Formula/vim.rb]: Please change this line to `depends_on "gtk+" => :optional`

File: Library/Formula/vim.rb [hunk 2]
```diff
@@ -15,6 +15,7 @@ class Vim < Formula
   option "override-system-vi", "Override system vi"
   option "disable-nls", "Build vim without National Language Support (translated messages, keymaps)"
   option "with-client-server", "Enable client/server mode"
+  option "with-gui-gtk2", "Build vim with GTK2 GUI"
```
Comment (sjackman) [inline Library/Formula/vim.rb]: Please delete this line (see below).

File: Library/Formula/vim.rb [hunk 3]
```diff
@@ -63,7 +65,11 @@ def install
     end
 
     opts << "--disable-nls" if build.include? "disable-nls"
-    opts << "--enable-gui=no"
+    if build.with? "gui-gtk2"
```
Comment (sjackman) [inline Library/Formula/vim.rb]: Please change this line to `if build.with? "gtk+"`

**Comment by sjackman** [comment]
Do you have access to a Mac OS machine? If this patch works on a Mac, it should be sent to Homebrew/homebrew rather than Homebrew/linuxbrew.

**Comment by paulhybryant** [comment]
I have updated the pull request.
This does not work on Mac OS because some files are not available in the gtk+ installation on Mac OS (gdkx.h to be specific).

**Comment by sjackman** [comment]
Merged. The preferred commit log format is:
`vim: Add option to build with gtk+ for Linuxbrew`

**Comment by sjackman** [comment]
Thanks for your contribution, Paul.

**Pull request closed without merging**